src/main/resources/schema/provider.definition.schema.v1.json

```
@@ -387,6 +387,10 @@
             "description": "A list of JSON pointers to properties (typically sensitive) that are able to be specified by the customer but unable to be returned in a Read request",
             "$ref": "#/definitions/jsonPointerArray"
         },
+        "conditionallyCreateOnlyProperties": {
+            "description": "A list of JSON pointers to properties that are only able to cause replacement during the Update request.",
```

think `only` might be misleading. Also, do you think it's worth adding an example? (need to scroll to see suggestion)
```suggestion
            "description": "A list of JSON pointers to properties that may or may not cause replacement during the Update request. For example, a version property might be upgradable in-place but downgrading may require resource replacement",
```

---

You think we should have a specific example like version?

---

@jotompki @wbingli @aygold92 @anshikg thoughts?

---

I would even give a specific example, like that for EBS volumes you can increase the `size` property, but decreasing it would require replacement https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/modify-volume-requirements.html#elastic-volumes-limitations

That being said, I would not even mention replacement, or if so mention that specifically as a cloudformation specific thing.  Cloud API will never replace your resource, it would just fail with `NotUpdatable` exception.  I would say something like:

```
A list of JSON pointers for properties that can only be updated under certain conditions.  For example, you can increase the size of an EBS volume but you cannot decrease it.  When updating this property for a resource in a CloudFormation stack, the resource will be replaced if it cannot be updated.
```

---

[One consideration about specific examples is that they may change in the future](https://github.com/awsdocs/aws-cloudformation-user-guide/pull/904). EBS might remove the restriction on decreasing `size` in-place eventually

---

true, I'm fine with the generic example you had above

---

how about using rds? db downgrade is unlikely to change over time:
```
A list of JSON pointers for properties that can only be updated under certain conditions.  For example, you can upgrade the engine version of a RDS DBInstance but you cannot downgrade it.  When updating this property for a resource in a CloudFormation stack, the resource will be replaced if it cannot be updated.
```